sfwebui.py

```diff
@@ -835,15 +851,25 @@ def startscan(self, scanname, scantarget, modulelist, typelist, usecase, cli=Non
             modlist.remove("sfp__stor_stdout")
 
         # Start running a new scan
-        scanId = sf.genScanInstanceGUID(scanname)
         if targetType in [ "HUMAN_NAME", "USERNAME" ]:
             scantarget = scantarget.replace("\"", "")
         else:
             scantarget = scantarget.lower()
-        p = mp.Process(target=SpiderFootScanner, args=(scanname, scantarget, targetType, scanId,
-                              modlist, cfg, modopts))
-        p.daemon = True
-        p.start()
+
+        try:
+            s = SpiderFootScanner(scanname, scantarget, targetType, modlist, cfg)
+            scanId = s.getId()
```

Comment:
Race

Reply:
Disagree. The ID is set during SpiderFootScanner initialisation. This line is called after the SpiderFootScanner object is created.